create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/conversion/CristinContributorExtractor.java

```diff
@@ -0,0 +1,103 @@
+
+    public static final String NAME_DELIMITER = ", ";
+    public static final String FIRST_NAME_CRISTIN_FIELD_NAME = "FirstName";
+    public static final String LAST_NAME_CRISTIN_FIELD_NAME = "LastName";
+    public static final String ORCID_FIELD_NAME = "orcid";
+
+    @JacocoGenerated
+    private CristinContributorExtractor() {
+    }
+
+    public static Contributor generateContributorFromCristin(Person person, AuthorTp authorTp,
+                                                             PersonalnameType correspondencePerson) {
+        return new Contributor(generateContributorIdentityFromCristinPerson(person),
+                               generateOrganizationsFromCristinAffiliations(person.getAffiliations()), Role.CREATOR,
+                               getSequenceNumber(authorTp), isCorrespondingAuthor(authorTp, correspondencePerson));
+    }
+
+    private static Identity generateContributorIdentityFromCristinPerson(Person cristinPerson) {
+        var identity = new Identity();
+        identity.setName(determineContributorName(cristinPerson));
+        identity.setOrcId(cristinPerson
+                              .getIdentifiers()
+                              .stream()
+                              .filter(CristinContributorExtractor::isOrcid)
+                              .findAny().map(TypedValue::getValue)
+                              .orElse(null));
+        identity.setId(cristinPerson.getId());
+        return identity;
+    }
+
+    private static List<Organization> generateOrganizationsFromCristinAffiliations(Set<Affiliation> affiliations) {
+        return affiliations
+                   .stream()
+                   .map(affiliation ->
+                            new Organization
+                                    .Builder()
+                                .withId(affiliation.getOrganization())
+                                .withLabels(affiliation.getRole().getLabels())
+                                .build())
+                   .collect(Collectors.toList());
+    }
+
+    private static int getSequenceNumber(AuthorTp authorTp) {
+        return Integer.parseInt(authorTp.getSeq());
+    }
+
+    private static String determineContributorName(Person person) {
+        return getLastName(person)
+               + NAME_DELIMITER
+               + getFirstName(person);
+    }
+
+    @NotNull
+    private static String getFirstName(Person person) {
+        return person.getNames().stream()
+                   .filter(CristinContributorExtractor::isFirstName).findFirst().map(
+                TypedValue::getValue).orElse(StringUtils.EMPTY_STRING);
+    }
+
+    @NotNull
+    private static String getLastName(Person person) {
+        return person.getNames().stream().filter(CristinContributorExtractor::isSurname).findFirst()
+                   .map(TypedValue::getValue).orElse(StringUtils.EMPTY_STRING);
+    }
+
+    private static boolean isFirstName(TypedValue typedValue) {
```

Comment:
TypedValue can never be null? Therefore safe to call its getType?

Reply:
Skal ordne nullsjekk